context_test.go

```diff
@@ -20,60 +22,48 @@ type ContextData struct {
 }
 
 func TestContextSaveLoad(t *testing.T) {
-	setContextDataPath("")
+	tmp, err := ioutil.TempDir("", "conode")
+	defer os.RemoveAll(tmp)
+	os.Setenv("CONODE_SERVICE_PATH", tmp)
+	initContextDataPath()
+
 	nbr := 10
 	c := make([]*Context, nbr)
 	for i := range c {
-		c[i] = createContext()
-	}
-	var wg sync.WaitGroup
-	wg.Add(nbr)
-	testLoadSave(t, true, c[0])
-	for i := range c {
-		go func(i int) {
-			testLoadSave(t, false, c[i])
-			wg.Done()
-		}(i)
+		c[i] = createContext(t)
 	}
-	wg.Wait()
 
-	tmp, err := ioutil.TempDir("", "conode")
-	log.ErrFatal(err)
-	defer os.RemoveAll(tmp)
-	os.Setenv("CONODE_SERVICE_PATH", tmp)
-	initContextDataPath()
+	testSaveFailure(t, c[0])
+
+	var wg sync.WaitGroup
 	wg.Add(nbr)
 	for i := range c {
 		go func(i int) {
-			testLoadSave(t, false, c[i])
+			testLoadSave(t, c[i])
 			wg.Done()
 		}(i)
 	}
 	wg.Wait()
+	log.Print(tmp)
```

Comment:
remove probably? or t.Log?

Reply:
Don't mix logging packages. If you think this might be useful, put something like
```
log.Lvl2("temp-directory is:", tmp)
```

Reply:
Forgot to delete that line, thanks!